geode-core/src/distributedTest/java/org/apache/geode/internal/cache/persistence/PersistentRecoveryOrderDUnitTest.java

```diff
@@ -1326,6 +1326,10 @@ public void testRecoverAfterConflict() {
     vm1.invoke(() -> getCache().close());
 
     vm0.invoke(() -> {
+      await().untilAsserted(
+          () -> assertThat(
+              InternalDistributedSystem.getAnyInstance().getCancelCriterion().cancelInProgress())
```

Comment:
Looking at the code related to cancelInProgress(); it looks like it may also be returning early, by checking isClosing(). I may be missing something...Can you please point me to the path where it shows/checks cacelInProgress is returns null after cache is completely closed...

Reply:
I think the purpose of the fix is to introduce some wait until vm0 completely closed the cache. Using cancelInProgress() is not easy to understand. 

I suggest to use:
        await().untilAsserted(()->assertThat(cache == null || cache.isClosed()));